ci-operator/step-registry/idp/external-oidc/keycloak/aks/set-dns/idp-external-oidc-keycloak-aks-set-dns-commands.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+if [[ "${USE_HYPERSHIFT_AZURE_CREDS}" == "true" ]] ; then
+    AZURE_AUTH_LOCATION="/etc/hypershift-ci-jobs-azurecreds/credentials.json"
+fi
+AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
+AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
+AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
+
+az --version
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+if [ ! -f ${SHARED_DIR}/keycloak-prefix ] ; then
+    echo "The keycloak-prefix file must be provided by a previous step!"
+    exit 1
+fi
+export KUBECONFIG="${SHARED_DIR}/kubeconfig"
+KEYCLOAK_PREFIX="$(cat ${SHARED_DIR}/keycloak-prefix)"
+KEYCLOAK_HOST="${KEYCLOAK_PREFIX}.${HYPERSHIFT_BASE_DOMAIN}"
+DNS_ZONE_RG_NAME="os4-common"
```

Comment:
Given ci-operator/step-registry/idp/external-oidc/keycloak/aks/set-dns/idp-external-oidc-keycloak-aks-set-dns-ref.yaml already sets the default of `DNS_ZONE_RG_NAME`, line 23 here uses hard code instead of trying to check the `DNS_ZONE_RG_NAME` set there. Is this expected?

Reply:
And, how about dynamically using the RG of the cluster instead of the team's preserved RG?